src/main/config/config_eeprom.c

```diff
@@ -268,28 +221,34 @@ static bool writeSettingsToEEPROM(void)
             // write the only instance
 	        record.flags |= CR_CLASSICATION_SYSTEM;
             config_streamer_write(&streamer, (uint8_t *)&record, sizeof(record));
+            chk = updateChecksum(chk, (uint8_t *)&record, sizeof(record));
             config_streamer_write(&streamer, reg->base, reg->size);
+            chk = updateChecksum(chk, reg->base, reg->size);
         } else {
-
             // write one instance for each profile
             for (uint8_t profileIndex = 0; profileIndex < MAX_PROFILE_COUNT; profileIndex++) {
                 record.flags = 0;
 
                 record.flags |= ((profileIndex + 1) & CR_CLASSIFICATION_MASK);
                 config_streamer_write(&streamer, (uint8_t *)&record, sizeof(record));
-
+                chk = updateChecksum(chk, (uint8_t *)&record, sizeof(record));
                 const uint8_t *base = reg->base + (reg->size * profileIndex);
                 config_streamer_write(&streamer, base, reg->size);
+                chk = updateChecksum(chk, base, reg->size);
             }
         }
     }
 
     configFooter_t footer = {
         .terminator = 0,
-        .chk = ~config_streamer_chk(&streamer),
     };
 
     config_streamer_write(&streamer, (uint8_t *)&footer, sizeof(footer));
+    chk = updateChecksum(chk, (uint8_t *)&footer, sizeof(footer));
+
+    // append checksum now
+    chk = ~chk;
+    config_streamer_write(&streamer, &chk, sizeof(chk));
```

Comment:
Old code did not checksum footer (ok with zero bytes and xor, but not when the algorithm is changed .. 

Maybe I can hide `config_streamer_write + updateChecksum` behind local macro to reduce duplication ... but checksum calculation should be separated from writer